R/pkg/R/sparkR.R

```diff
@@ -655,6 +655,45 @@ sparkCheckInstall <- function(sparkHome, master, deployMode) {
     } else {
       if (interactive() || isMasterLocal(master)) {
         message("Spark not found in SPARK_HOME: ", sparkHome)
+        # If EXISTING_SPARKR_BACKEND_PORT environment variable is set, assume
+        # that we're in Spark submit. spark-submit always sets Spark home
+        # so this case should not happen. This is just a safeguard.
+        isSparkRSubmit <- Sys.getenv("EXISTING_SPARKR_BACKEND_PORT", "") != ""
```

Comment:
This is the same way as above:
https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/R/pkg/R/sparkR.R#L153-L155